src/pydantic_cereal/main.py

```diff
@@ -48,15 +49,42 @@ class CerealContext(AbstractContextManager):
     This is managed by the [`Cereal`][pydantic_cereal.Cereal] class - users shouldn't use this directly!
     """
 
-    def __init__(self, cereal: "Cereal", workdir: Union[UPath, Path, str]) -> None:
+    def __init__(
+        self,
+        cereal: "Cereal",
+        target_path: Union[UPath, Path, str],
+        fs: Optional[AbstractFileSystem] = None,
+    ) -> None:
         assert isinstance(cereal, Cereal)
-        self._workdir = UPath(workdir)
+        if fs:
+            assert isinstance(target_path, str)
+            inner_path = target_path
+        else:
+            if isinstance(target_path, str):
+                fs, _, paths = get_fs_token_paths(target_path)
+                inner_path = paths[-1]
+            elif isinstance(target_path, Path) and not isinstance(target_path, UPath):
+                fs, _, paths = get_fs_token_paths(f"file:{target_path}")
```

Comment:
Is this the right thing to prepend? Shouldn't it be `f"file:/{target_path}"` or `f"file://{target_path}"`?

Yes, number of slashes becomes quite confusing...

Reply:
The `f"file:{target_path}` also works, but you're right - its better to be consistent.
```suggestion
                fs, _, paths = get_fs_token_paths(f"file://{target_path}")
```